packages/ipfs-core/src/components/repo/index.js

```diff
@@ -0,0 +1,29 @@
+'use strict'
+
+const createGC = require('./gc')
+const createStat = require('./stat')
+const createVersion = require('./version')
+
+class RepoAPI {
+  /**
+   * @param {Object} config
+   * @param {GCLock} config.gcLock
+   * @param {Pin} config.pin
+   * @param {Repo} config.repo
+   * @param {Refs} config.refs
+   */
+  constructor ({ gcLock, pin, repo, refs }) {
+    this.gc = createGC({ gcLock, pin, refs, repo })
+    this.stat = createStat({ repo })
+    this.version = createVersion({ repo })
```

Comment:
maybe we should inline the code for these in this file reducing the boilerplate, could be done for other API classes also.

Reply:
Not sure I understand what you mean, could you please elaborate ?

Reply:
now that we have the wrapper classes maybe we dont need one file per function everywhere and could inline those functions inside the wrapper class file reducing the types boilerplate.

Reply:
I don't mind that, but would rather pursue as separate endeavour if we decide to do it.